compiler/rustc_infer/src/infer/error_reporting/mod.rs

```diff
@@ -1922,7 +1922,8 @@ impl<'tcx> TypeErrCtxt<'_, 'tcx> {
                         (ty::Uint(ty::UintTy::U8), ty::Char) => {
                             if let Ok(code) = self.tcx.sess().source_map().span_to_snippet(span)
                                 && let Some(code) = code.strip_prefix('\'').and_then(|s| s.strip_suffix('\''))
-                                && code.chars().next().map_or(false, |c| c.is_ascii())
+                                && code.chars().count() < 5 // allows escapes other than Unicode escapes
```

Comment:
don't we just need to check there's a `\u` at the beginning? I don't like using the length as a heuristic here.

Reply:
Fair, I'm just lazy, I have changed it to check !start_with("\\u") - is that more sensible ?